src/trace/context.ts

```diff
@@ -135,13 +135,11 @@ export function sendXraySubsegment(segment: string) {
     client = createSocket("udp4");
     // Send segment asynchronously to xray daemon
     client.send(message, 0, message.length, port, address, (error, bytes) => {
+      client?.close();
       logDebug(`Xray daemon received metadata payload`, { error, bytes });
     });
   } catch (error) {
     logDebug("Error occurred submitting to xray daemon", { error });
```

Comment:
Should we close the socket inside the catch as well? Make sense to close it inside the callback in the happy path, but in the case where an exception is thrown I think closing the socket would make sense as well.

Reply:
I've re-added this client.close